libvast/src/system/partition_transformer.cpp

```diff
@@ -25,10 +25,9 @@ namespace vast::system {
 // constructed, we don't have to spawn separate indexer actors and
 // stream data but can just compute everything inline here.
 void partition_transformer_state::add_slice(const table_slice& slice) {
-  const auto& layout = slice.layout();
+  const auto& layout = flatten(slice.layout());
```

Comment:
I've been trying to get rid of flatten calls as part of #1888. Can you add a comment that explains why we would need to flatten here so it's easier for me to reason about this change when updating this to make use of the new type system?